packages/@ember/-internals/glimmer/lib/helpers/get.ts

```diff
@@ -90,55 +95,39 @@ import { referenceFromParts } from '../utils/references';
   @since 2.1.0
  */
 export default function(args: VMArguments) {
-  let sourceReference = args.positional.at(0);
-  let pathReference = args.positional.at(1);
+  let sourceRef = args.positional.at(0);
+  let pathRef = args.positional.at(1);
 
-  let path = pathReference.value();
-
-  if (pathReference.isConst()) {
+  if (isConstRef(pathRef)) {
```

Comment:
So, not at this point, because we have no guarantee that `pathRef` and `sourceRef` have been executed at least once. If they are primitives (e.g. `{{get "foo" "length"}}`) then they will be, but that's a fairly rare case. The `pathRef` being a primitive is much more common though, which is why we check it.

Basically, primitives are guaranteed to be constant at first, and Compute refs need to be calculated at least once. We don't want to pull on a ref until it's actually going to be used, so there are cases where it doesn't make sense to check.